bpmn-model/src/test/java/io/camunda/zeebe/model/bpmn/validation/ZeebeEscalationValidationTest.java

```diff
@@ -0,0 +1,280 @@
+/*
+ * Copyright © 2017 camunda services GmbH (info@camunda.com)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.camunda.zeebe.model.bpmn.validation;
+
+import static io.camunda.zeebe.model.bpmn.validation.ExpectedValidationResult.expect;
+
+import io.camunda.zeebe.model.bpmn.Bpmn;
+import io.camunda.zeebe.model.bpmn.BpmnModelInstance;
+import io.camunda.zeebe.model.bpmn.builder.AbstractThrowEventBuilder;
+import io.camunda.zeebe.model.bpmn.builder.SubProcessBuilder;
+import io.camunda.zeebe.model.bpmn.instance.BoundaryEvent;
+import io.camunda.zeebe.model.bpmn.instance.CallActivity;
+import io.camunda.zeebe.model.bpmn.instance.Escalation;
+import io.camunda.zeebe.model.bpmn.instance.EscalationEventDefinition;
+import io.camunda.zeebe.model.bpmn.instance.Process;
+import io.camunda.zeebe.model.bpmn.instance.SubProcess;
+import java.util.function.Consumer;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+
+class ZeebeEscalationValidationTest {
+
+  private BpmnModelInstance process(final Consumer<SubProcessBuilder> builder) {
+    return Bpmn.createExecutableProcess().startEvent().subProcess("sp", builder).endEvent().done();
+  }
+
+  @Test
+  @DisplayName(
+      "An escalation boundary event should only allow attaching to sub-process or call-activity")
+  void verifyEscalationBoundaryEventOnlyAllowAttachingToSubProcessOrCallActivity() {
```

Comment:
🔧 We verify here that we cannot attach it to something that isn't a subprocess or call activity. Let's also introduce 2 tests to verify that we can attach it to a subprocess and call activity and that the process will be valid. This is indirectly already covered by the other test cases, but this makes it easier to spot where the error is when we break something and this validation fails.